feat: Document the global cache fallback during development (#527)
serverpod/serverpod_docs: docs/06-concepts/08-caching.md

}
```

In total, there are three caches where you can store your objects. Two caches are local to the server handling the current session, and one is distributed across the server cluster through Redis. There are two variants for the local cache: one regular cache, and a priority cache. Place objects that are frequently accessed in the priority cache.

Depending on the type and number of objects that are cached in the global cache, you may want to specify custom caching rules in Redis. This is currently not handled automatically by Serverpod.

:::info

> During development, you can run and test uses of the global cache on your code without depending on a running Redis server. If the server runs on `development`/`test` run mode and the Redis cache is configured, but not available, the global cache will fall back to an isolated in-memory cache.

Reviewer: Three accuracy points and small wording:
- Fallback fires in any non-production mode, not just `development`/`test` (`lib/src/server/serverpod.dart. around 807`).
- Also fires when Redis isn't configured, not only when configured-but-unavailable (`caches.dart around 25-28`).
- Wording: `runs on` → `runs in`, drop the comma in "configured, but not available", `uses of the global cache on your code` → `code that uses the global cache`.

```suggestion
During development, you can run code that uses the global cache without a running Redis server. If Redis is unreachable (or not configured) in a non-production run mode, the global cache falls back to an isolated in-memory cache.
```
